napari/_app/actions/_layer_actions.py

```diff
@@ -0,0 +1,162 @@
+        callback=_layer_actions._convert_to_labels,
+        enablement=(
+            (
+                (LLCK.num_selected_image_layers >= 1)
+                | (LLCK.num_selected_shapes_layers >= 1)
+            )
+            & LLCK.all_selected_layers_same_type
+        ),
+        menus=[LAYERCTX_CONVERSION],
+    ),
+    Action(
+        id=CommandId.LAYER_CONVERT_TO_IMAGE,
+        title=CommandId.LAYER_CONVERT_TO_IMAGE.title,
+        callback=_layer_actions._convert_to_image,
+        enablement=(
+            (LLCK.num_selected_labels_layers >= 1)
+            & LLCK.all_selected_layers_same_type
+        ),
+        menus=[LAYERCTX_CONVERSION],
+    ),
+    Action(
+        id=CommandId.LAYER_MERGE_STACK,
+        title=CommandId.LAYER_MERGE_STACK.title,
+        callback=_layer_actions._merge_stack,
+        enablement=(
+            (LLCK.num_selected_layers > 1)
+            & (LLCK.num_selected_image_layers == LLCK.num_selected_layers)
+            & LLCK.all_selected_layers_same_shape
+        ),
+        menus=[LAYERCTX_SPLITMERGE],
+    ),
+    Action(
+        id=CommandId.LAYER_TOGGLE_VISIBILITY,
+        title=CommandId.LAYER_TOGGLE_VISIBILITY.title,
+        callback=_layer_actions._toggle_visibility,
+        menus=[
+            {
+                'id': MenuId.LAYERLIST_CONTEXT,
+                'group': MenuGroup.NAVIGATION,
+            }
+        ],
+    ),
+    Action(
+        id=CommandId.LAYER_LINK_SELECTED,
+        title=CommandId.LAYER_LINK_SELECTED.title,
+        callback=_layer_actions._link_selected_layers,
+        enablement=(
+            (LLCK.num_selected_layers > 1) & ~LLCK.num_selected_layers_linked
+        ),
+        menus=[{**LAYERCTX_LINK, 'when': ~LLCK.num_selected_layers_linked}],
+    ),
+    Action(
+        id=CommandId.LAYER_UNLINK_SELECTED,
+        title=CommandId.LAYER_UNLINK_SELECTED.title,
+        callback=_layer_actions._unlink_selected_layers,
+        enablement=LLCK.num_selected_layers_linked,
+        menus=[{**LAYERCTX_LINK, 'when': LLCK.num_selected_layers_linked}],
+    ),
+    Action(
+        id=CommandId.LAYER_SELECT_LINKED,
+        title=CommandId.LAYER_SELECT_LINKED.title,
+        callback=_layer_actions._select_linked_layers,
+        enablement=LLCK.num_unselected_linked_layers,
+        menus=[LAYERCTX_LINK],
+    ),
+]
+
+
+for _dtype in (
+    'int8',
+    'int16',
+    'int32',
+    'int64',
+    'uint8',
+    'uint16',
+    'uint32',
+    'uint64',
+):
+    cmd: CommandId = getattr(CommandId, f'LAYER_CONVERT_TO_{_dtype.upper()}')
+    LAYER_ACTIONS.append(
+        Action(
+            id=cmd,
+            title=cmd.title,
+            callback=partial(_layer_actions._convert_dtype, mode=_dtype),
+            enablement=(_only_labels & (LLCK.active_layer_dtype != _dtype)),
+            menus=[{'id': MenuId.LAYERS_CONVERT_DTYPE}],
+        )
+    )
+
+_image_is_3d = (LLCK.active_layer_type == "image") & LLCK.active_layer_ndim > 2
```

Comment:
I really like having these useful little pieces of context separate from the action definition. Would it be better if these were all in one place near the top of the module? It would put useful context creation examples in the same place, might promote reuse from plugins etc

Reply:
sure, makes sense to me. will do